### ISSUE TITLE
Db structure!

### ISSUE THREAD

**Issue by RonaldColyar** [FotieMConstant/bond#37]
Assuming we are just going to continue using firebase: 
# Users
- DisplayName
- Username
- Groups(priority groups  like family , lovers,friends etc) the groups will contain the usernames of the people and its priority. The priority will be normal , above normal , and number 1.
- Sounds
- Contacts(a list of usernames)
- Received sounds (the sounds that others sent to you! This will just be a link to the sound in the other person's  storage , but you may think this is an issue because it seem like users can read each other's data. We can solve this issue by making rules that limit users to only reading a specfic portion of the user's data model) 
- Contact Requests.

**Comment by RonaldColyar** [comment]
@FotieMConstant

**Comment by RonaldColyar** [comment]
@quiet-programmer

**Comment by FotieMConstant** [comment]
Are we finally sticking to firebase? Because I think we should get something we can have more control on like I talked with @RonaldColyar. Both now and later

**Comment by RonaldColyar** [comment]
@FotieMConstant it could work well.

**Comment by RonaldColyar** [comment]
We could collect our data and have a separate database also

**Comment by RonaldColyar** [comment]
But not the same data that we store in firebase it would be redundant,

**Comment by FotieMConstant** [comment]
Yeah, i think so as well. Please does anyone has Postgre or Mongo db online we can use?

**Comment by dilanlongla** [comment]
@FotieMConstant we could configure a postgre db on heroku but it will be limited as they grant a max of 20 simultenous connections to the db

**Comment by FotieMConstant** [comment]
How much does it cost if we want to upgrade! Any idea?

**Comment by dilanlongla** [comment]
@FotieMConstant the pricing plan varies greatly, here is a link you can check it out 
[postgreSQL db Heroku pricing plan](https://elements.heroku.com/addons/heroku-postgresql)

**Comment by dilanlongla** [comment]
### **Relational Database Structure**
@FotieMConstant @RonaldColyar  before we join anything i'll share the structure of the db here so we could discuss it

- **Here we have the class diagram of the database**

![class diagram](https://user-images.githubusercontent.com/44490676/114344546-f0d6c880-9b57-11eb-8e74-6179a8652a1b.PNG)


- **Here have the Physical datamodel** 

![pdm](https://user-images.githubusercontent.com/44490676/114344738-47dc9d80-9b58-11eb-8b75-7ce2e97a8ebb.PNG)

NB : _I used the information i had in mind on the project and that which @RonaldColyar shared recently to come up with this database_
